### DISCUSSION TITLE
CMT2300A

### DISCUSSION THREAD

**Discussion by ziomal1234** [tbnobody/OpenDTU#812] [General]
How connect CMT2300A. I have hms2250

**Comment by Solarteur** [comment]
Please see this discussion:
https://github.com/tbnobody/OpenDTU/discussions/336

**Comment by ziomal1234** [comment]
[v23.4.15](https://github.com/tbnobody/OpenDTU/releases/tag/v23.4.15)
"This is the first version which supports the new HMS/HMT inverters. The documentation is still not perfect, so maybe you have to know what you are doing :-) Further improvements will come."

**Comment by tbnobody** [comment]
Please have a look at the files https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles/blinkyparts_esp32.json and https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles/nodemcu_esp32.json for a sample pin assignment. I am currently testing with the assignment in the blinkyparts pin definition file. it uses the same pins as the nrf module (cannot be used in parallel). the assignment in the nodemcu file is able to use both modules in parallel

**Reply by ziomal1234** [reply]
It worki with hms,2250

**Comment by ziomal1234** [comment]
it will work with hms2250 ??

**Comment by ziomal1234** [comment]
do you have a diagram ?

**Comment by ziomal1234** [comment]
where can i buy EBYTE E49-900M20S
with shipping to Poland

**Comment by Solarteur** [comment]
For example here: https://de.aliexpress.com/item/1005003290274214.html?spm=a2g0o.productlist.main.47.480030f273bJsY&algo_pvid=aca58083-a120-45c0-930a-86a1b4863a81&algo_exp_id=aca58083-a120-45c0-930a-86a1b4863a81-23&pdp_npi=3%40dis%21EUR%212.63%212.6%21%21%21%21%21%402100bfe316816606154453289d0703%2112000025045805357%21sea%21DE%210&curPageLogUid=GYWOkT0S87c3

**Comment by ziomal1234** [comment]
Danke.
Hat jemand versucht, wie es funktioniert?
Hast du einen Film?

**Comment by ziomal1234** [comment]
Sie haben ein fertiges Layout zum Verkauf

**Comment by Solarteur** [comment]
Ich habe einen HM-Wechselrichter und setze ein anderes Funkmodul ein. Ich habe auch kein Layout für eine Platine.

**Comment by ziomal1234** [comment]
Ich habe hms2250

**Comment by ziomal1234** [comment]
"cmt": {
            "clk": 18,
            "cs": 4,
            "fcs": 5,
            "sdio": 23,
            "gpio2": 19,
            "gpio3": 16
        },
![Bez tytułu](https://user-images.githubusercontent.com/74967995/232382596-998bb530-89ad-486a-8020-5b3101f24cad.jpg)

can you tell me which pins to connect to each other?

**Comment by Yosh1977** [comment]
Hallo,

erstmal nen schönen Dank für die super Arbeit.

https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles/nodemcu_esp32.json

Ich habe dazu ne Frage. Und zwar ist es ja mit dem File möglich beide Funkmodule anzusteuern, aber bei den gpio2 und gpio3 steht -1. Werden diese nicht gebraucht? Oder was soll mir das sagen? In dem File wo nur das eine Modul läuft, sind die beiden gpio2 und gpio3 auf dem Esp verbunden.
Sind beide Funkmodul mit Display möglich?
Gruß

**Comment by tbnobody** [comment]
> Und zwar ist es ja mit dem File möglich beide Funkmodule anzusteuern, aber bei den gpio2 und gpio3 steht -1.

Die  gpio2 und gpio3 sind optional (siehe auch https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles.md)
Wenn die Pins nicht belegt sind, wird der Status über den SPI Bus abgerufen.

> Sind beide Funkmodul mit Display möglich?

Ich muss zugeben ich habe diesen Fall noch nicht aktiv getestet. Aber da die Config ja Pin 21 und 22 nicht belegt würde ich sagen, dass zumindest die OLED Displays gehen sollten (i²c Bus). Das Nokia glaub ich funktioniert aktuell damit nicht.

**Reply by Yosh1977** [reply]
danke

**Reply by Yosh1977** [reply]
Da du das ja schon getestet hast, läuft das genausogut wie mit den NRF Modulen ?

**Reply by tbnobody** [reply]
Die CMT Module haben meiner Meinung nach einen viel besseren Empfang (868MHz statt 2,4GHz). Ebenso ist die ganze Kanal Auswahl stabiler. Den Parallelbetrieb hab aus beiden Modulen hab ich persönlich noch nicht getestet, aber ich  weiß von $Leuten, dass es funktioniert.

**Reply by Yosh1977** [reply]
besten dank
